src/services/services.ts

```diff
@@ -1489,6 +1472,23 @@ namespace ts {
             };
         }
 
+        function shouldGetType(sourceFile: SourceFile, node: Node, position: number): boolean {

```

Comment:
this is purely stylistic. but i think it read better before. in general refactoring inner pieces of logic that are rather specific to a certain code path makes it harder to understand the flow.

Reply:
Previously we used switch-case fallthrough to determine whether to reach the last case. That was possible with only two cases, but it's not possible any more when we need two cases to both goto the same third case.